### ISSUE TITLE
Allow glob pattern for project path(s)

### ISSUE THREAD

**Issue by thiagomajesk** [formulahendry/vscode-dotnet-test-explorer#66]
Related to #12 (that was closed apparently without any reason)...

The proposal is simple: dotnet-test-explorer be able to pick up test projects from one or more specified glob patterns. Eg.: `dotnet-test-explorer.testProjectPath: ["*.Tests", "tests/**"]`.

Today if you leave it blank, the extension will try to run `dotnet tests` on all projects in the root folder, polluting the console with error messages from non-test projects.

**Comment by AlexTeixeira** [comment]
Hi,

I investigate more about this, and found multiple pieces of code that are related to this.

![testexplorer2](https://user-images.githubusercontent.com/6555080/39517414-1200d4ea-4e00-11e8-97f9-135dc6d0d085.PNG)
![testexplorer](https://user-images.githubusercontent.com/6555080/39517415-122ecd46-4e00-11e8-9f80-0b8ff05732c4.PNG)

The definition of `testProjectPath` is in the package json. If I have some time, I will try to push a PR.

I do not see any rules about pushing PR for this project. If someone is aware of that, please notify.

**Comment by teamdynamiq** [comment]
Discovery of the tests are done by executing dotnet test (either using workspace root or using testDirectoryPath if that setting is used). As far as I remember there was no way to give dotnet test a glob pattern but only a single .sln file or a single csproj file. So I don't think it's a quick fix given the way the extension is organized today.

But you touch on something else @thiagomajesk I've been thinking about. That the terminal output is cluttered and it's usually somewhat hard to find the info I want (which basically is which test failed and why) and that is something we should be able to do better.

**Comment by AlexTeixeira** [comment]
@thiagomajesk, I currently trying to change the code base on a fork. And updating the path is store here in `testCommand.ts` in `private testDirectoryPath: string;`

Changing it into a `string[]` impact several methods and the impact in the extensions can be important.
I'm currently trying something but I'm new on developping extension for VsCode so I lost some time looking how I can test my changes.

**Comment by AlexTeixeira** [comment]
@stefanforsberg  Yeah i miss it... Currently it seems my changes haven't generated any regression. But for two projects, the result still empty.

**Comment by AlexTeixeira** [comment]
@thiagomajesk, @teamdynamiq. I have something working for multiple projects. This is quite simple now, multiple path without any folder sub folder discovering.

Here the changes : 
https://github.com/AlexTeixeira/vscode-dotnet-test-explorer/commit/b2669a4868ef7f07559721dfc46136375a681997

![testexplorer](https://user-images.githubusercontent.com/6555080/39537920-578f22d8-4e3b-11e8-9775-53bc5e642947.PNG)

Now I need to work in the execution flow because the first project do not keep the test results

**Comment by blackzky** [comment]
@AlexTeixeira, how I can I try your solution?

**Comment by AlexTeixeira** [comment]
@blackzky, yes I added you as a collaborator.
I will try to do more tests this week end and add unit tests for multiple project. When is done I will submit a PR
